### ISSUE TITLE
Dogfooding Waku v2 through mini chat app

### ISSUE THREAD

**Issue by oskarth** [status-im/nim-waku#399]
# Problem

We want to dogfood Waku v2 to find any issues that might exist with it and to work out kinks. While there is ongoing work to do this with Stimbus/Desktop/App, this is taking a bit longer than expected and is quite complex. How can we dogfood Waku as early as possible?

# Suggested solutions

Luckily, we don't need the whole Status protocol or app to do basic dogfooding. We just need a basic chat interface (provided for by e.g. `chat2` protocol and a single "chat topic" to replace #waku conversations. We don't need more complex encryption features, and things like nicknames can simply be hardcoded.

# Acceptance criteria

`chat2` app good enough to replace daily use for #waku conversations. This means: can send messages and receive, and fetch messages received while gone. Also some basic instructions for how to start a node and join chat.

# Enhancements

Interop with JS client to guide Waku JS critical path.

Obviously Status desktop use but that's out of scope for this issue and repo ;)

**Comment by oskarth** [comment]
Running `chat2` it seems to work but need second node to confirm. Probably need to clean up the logs to make UX a bit nicer (pipe to logfile and leave STDOUT for chat?).

**Comment by oskarth** [comment]
Can confirm that `./build/chat2 --storenode:/ip4/134.209.139.210/tcp/30303/p2p/16Uiu2HAmPLe7Mzm8TsYUubgCAW1aJoeFScxrLj8ppHFivPo97bUZ --staticnode:/ip4/134.209.139.210/tcp/30303/p2p/16Uiu2HAmPLe7Mzm8TsYUubgCAW1aJoeFScxrLj8ppHFivPo97bUZ` works for three of us, including historical messages.

What do we need for this to replace current chat? Sketch:

- logs out of stdout
- commandline flag nick
- not hardcoded store/staticnodes (?)

- maybe use random of cluster node? (otherwise a bit odd we all connect to same node)
- thoughts on how the peer(s) selection should work by default here?

Discord<>Wakuv2 bridge would be nice too, e.g. using existing chat2 STDIN/OUT, or Nim JSON RPC API.

**Comment by jm-clius** [comment]
Agreed, @oskarth. Off the top of my head I'd add:
1. some timestamps
2. better chat prompts

> * maybe use random of cluster node? (otherwise a bit odd we all connect to same node)
> * thoughts on how the peer(s) selection should work by default here?

We could scrape the list of available test nodes from fleets.status.im and just select one? Selecting at random is probably a decent approach for now, as we have no proper scoring/priority mechanisms (yet).

**Comment by staheri14** [comment]
> We could scrape the list of available test nodes from fleets.status.im and just select one? Selecting at random is probably a decent approach for now, as we have no proper scoring/priority mechanisms (yet).

I agree with the random selection, later we may select the node with the closest ip address or ping the cluster nodes and select the one with the least rtt

**Comment by jm-clius** [comment]
So, thanks to [Jakub's script](https://github.com/status-im/nim-waku/issues/408) that peers fleet nodes, we should now have a "fully" functional `chat2` app. To participate simply run `./build/chat2`, or, if you want history, `./build/chat2 --store=true`. Peer selection is automatic.  Timestamps should be in local time.
Let me know if you spot any bugs or have UX concerns.

cc @D4nte

**Comment by oskarth** [comment]
Update: mostly blocked by https://github.com/status-im/nim-waku/issues/404 I believe

Some store consistency things would improve experience here too

**Comment by jm-clius** [comment]
> Update: mostly blocked by #404 I believe

Correct. #404 is currently WIP, but on hold while dealing with some higher priority bugs (e.g. #480). Will hopefully get some traction this week.

**Comment by D4nte** [comment]
> So, thanks to [Jakub's script](https://github.com/status-im/nim-waku/issues/408) that peers fleet nodes, we should now have a "fully" functional `chat2` app. To participate simply run `./build/chat2`, or, if you want history, `./build/chat2 --store=true`. Peer selection is automatic. Timestamps should be in local time.
> Let me know if you spot any bugs or have UX concerns.
> 
> cc @D4nte

Ok I'll add a todo for a similar peer selection on js-chat side which would also allow dog fooding for js-waku.

**Comment by oskarth** [comment]
Largely done, will open specific issues as needed
